src/python/pants/util/memo.py

```diff
@@ -269,22 +269,6 @@ def memoized_classproperty(
     )
 
 
-def memoized_staticmethod(
```

Comment:
Lmk if I should keep this. We weren't using it, and it seems that we generally have been using class properties rather than static properties.

```
▶ rg '@static' | wc -l
     117

▶ rg '@class' | wc -l
     401
```

Reply:
This was just added for some idea of completeness, and so people have a clear way to add memoization to existing `@staticmethod`s without having to hack it in themselves or risk breaking changes by turning it into a `@memoized_classmethod` or something.